File: src/test/java/boj/problems/step2/No9498Test.java
package boj.problems.step2;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.CsvSource;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.StringReader;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class No9498Test {
    @DisplayName("시험 성적 테스트")
    @ParameterizedTest
    @CsvSource(value = {"100:A", "90:A", "89:B", "80:B", "79:C", "70:C", "69:D", "60:D", "59:F", "0:F"}, delimiter = ':')
    void test(String given, String expected) throws IOException {
        System.out.println("시험 성적 : https://www.acmicpc.net/problem/9498");

        BufferedReader br_given = new BufferedReader(new StringReader(given));

        assertEquals(expected, No9498.solve(br_given));

        br_given.close();
    }
}

File: src/test/kotlin/boj/problems/step2/No9498Test.kt
package boj.problems.step2

import io.kotest.core.spec.style.StringSpec
import io.kotest.matchers.shouldBe

class No9498Test : StringSpec({
    "시험 성적 테스트" {
        val testCases = listOf(
            "100" to "A",
            "90" to "A",
            "89" to "B",
            "80" to "B",
            "79" to "C",
            "70" to "C",
            "69" to "D",
            "60" to "D",
            "59" to "F",
            "0" to "F"
        )

        testCases.forEach { (given, expected) ->
            No9498.solve(given.byteInputStream().bufferedReader()) shouldBe expected
        }
    }
})
